### PR TITLE
ReadRows integration test

### LINKED ISSUE #77: Implement integration test for ReadRows() against the Cloud Bigtable Emulator
At the end of this task we will have an integration test for `Table::ReadRows()` against the Cloud Bigtable Emulator.  This should be part of the CI builds.  Look at #74 if you need inspiration.

### PR DESCRIPTION
**Pull request by DanielMahu**
This fixes #77.

I tried to cover the basic cases in a style similar to the other tests
- all rows, including a large key and large value
- a subset of rows
- no rows
- error - inexistent table.

### CODE REVIEW

**Comment by DanielMahu** [comment]
During testing I found a couple of corner cases with the empty `RowSet` that seem broken (they're not in the code right now):

This one failed against the emulator, but succeeded in production, when added to TableReadRowsAllRows:
  ```c++
  auto read4 =
    table->ReadRows(bigtable::RowSet(), bigtable::Filter::PassAllFilter());
  CheckEqualUnordered(created,  MoveCellsFromReader(read4));
  ```
it's testing that the default `RowSet` matches all rows, which I think is the documented behaviour.

This failed in both, in TableReadRowsNoRows:
  ```c++
  ASSERT_TRUE(bigtable::RowSet(bigtable::RowRange::Empty()).IsEmpty());
  auto read3 = table->ReadRows(bigtable::RowSet(bigtable::RowRange::Empty()),
                               bigtable::Filter::PassAllFilter());
  CheckEqualUnordered(expected, MoveCellsFromReader(read3));
  ```
and was broken because we'd expect that reading an empty `RowSet` should produce no results, but instead it returned all the rows. Maybe we should test for `IsEmpty()` and skip the actual read if it is true?

**Comment by coryan** [comment]
Thanks for the tests!  For `read4`, if the emulator does not match production please file a bug with the emulator (check #151 for an existing list), skip the test when running against the emulator, and add a TODO for it.

For `read3`, is there a way to request an empty row set to the production server?  If so, what do you think if we issue that request when `IsEmpty()` returns true?

File: bigtable/client/testing/table_integration_test.cc [hunk 1]
```diff
@@ -71,6 +71,16 @@ std::vector<bigtable::Cell> TableIntegrationTest::ReadRows(
   return result;
 }
 
+std::vector<bigtable::Cell> TableIntegrationTest::MoveCellsFromReader(
+    bigtable::RowReader& reader) {
+  std::vector<bigtable::Cell> result;
+  for (auto const& row : reader) {
+    std::copy(row.cells().begin(), row.cells().end(),
```
Comment (coryan) [inline bigtable/client/testing/table_integration_test.cc]: Consider using `std::move()` (the function in `<algorithm>` not the function in `<utility>`)

Comment (coryan) [reply bigtable/client/testing/table_integration_test.cc]: And if you think it is not a good idea or not possible tell me here!

Comment (DanielMahu) [reply bigtable/client/testing/table_integration_test.cc]: done.

**Comment by coryan** [comment]
@DanielMahu any news?

File: bigtable/tests/data_integration_test.cc [hunk 2]
```diff
@@ -211,6 +215,130 @@ TEST_F(DataIntegrationTest, TableReadRowNotExistTest) {
...
+
+TEST_F(DataIntegrationTest, TableReadRowsWrongTable) {
+  std::string const table_name = "table-read-rows-wrong-table";
+  bigtable::Table table(data_client_, table_name);
+
+  auto read1 =
+      table.ReadRows(bigtable::RowSet(bigtable::RowRange::InfiniteRange()),
+                     bigtable::Filter::PassAllFilter());
+#if GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
+  EXPECT_THROW(read1.begin(), std::runtime_error);
```
Comment (coryan) [inline bigtable/tests/data_integration_test.cc]: I think I know why this fails with no exceptions enabled.  Calling `read1.begin()` is an error, but we no longer crash the application when the iterator first fails.  The application needs to call `read1.Finish()` to get the error status, and only if they fail to retrieve the status does the code crash (to avoid silent errors).

Comment (DanielMahu) [reply bigtable/tests/data_integration_test.cc]: Thanks for pointing that out. But I actually got confused by the line here https://github.com/GoogleCloudPlatform/google-cloud-cpp/blob/dc5c782d69c8f455f075ba602aa90183fbb427c5/bigtable/client/table.cc#L74
that always sets `raise_on_error` to true... and I still am, it seems to me it's still going down that code path in the `noex::` namespace?

I reran the test, with
```cpp
  EXPECT_EQ(read1.begin(), read1.end());
```
and got this:

```
Aborting because exceptions are disabled: Unretriable error: table "projects/emulated-1524060748/instances/data-test/tables/table-read-rows-wrong-table" not found
```
so the program does die.

Is the no-exception code path ready? otherwise we could submit without this test and fix it separately together with that.

Comment (DanielMahu) [reply bigtable/tests/data_integration_test.cc]: I've changed noex::Table::ReadRows to use raise_on_error=false, I think that should do the trick. PTAL
